### PR TITLE
[MJLINK-88] separate Windows and Unix command line tests

### PR DESCRIPTION
**Pull request by elharo**
This likely doesn't actually fix the plugin on Windows, but does let tests pass so unrelated changes are not blocked. It also adds tests that show how the plugin operates on Windows so future PRs can build on that.

### CODE REVIEW

**Comment by michael-o** [comment]
Bah, ugly Plexus command line wrapper.

**Comment by elharo** [comment]
One day we'll be able to use ToolProvider to avoid this sort of thing. In the meantime, can I get an approval on this PR?

File: src/test/java/org/apache/maven/plugins/jlink/JLinkMojoTest.java [hunk 1]
```diff
@@ -45,12 +51,8 @@ void double_quote_every_argument() throws Exception {
...
 
     @Test
-    void single_quotes_shell_command() throws Exception {
-        // given
-        JLinkMojo mojo = new JLinkMojo();
-        Field stripDebug = mojo.getClass().getDeclaredField("stripDebug");
-        stripDebug.setAccessible(true);
-        stripDebug.set(mojo, Boolean.TRUE);
+    void single_quotes_shell_command_unix() throws Exception {
+        Assumptions.assumeFalse(System.getProperty("os.name").startsWith("Windows"));
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/plugins/jlink/JLinkMojoTest.java:55]: There is an annotation `@DisabledOnOs`

File: src/test/java/org/apache/maven/plugins/jlink/JLinkMojoTest.java [hunk 2]
```diff
@@ -61,4 +63,19 @@ void single_quotes_shell_command() throws Exception {
                 .isEqualTo(
                         "/bin/sh -c '/path/to/jlink \"--strip-debug\" \"--module-path\" \"foo:bar\" \"--add-modules\" \"mvn,jlink\"'");
     }
+
+    @Test
+    void single_quotes_shell_command_windows() throws Exception {
+        Assumptions.assumeTrue(System.getProperty("os.name").startsWith("Windows"));
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/plugins/jlink/JLinkMojoTest.java:69]: `@EnabledOnOs`

Comment (elharo) [reply src/test/java/org/apache/maven/plugins/jlink/JLinkMojoTest.java]: Neat. Haven't seen that one  before. Done.

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**